manual/src/cmds/runtime-tracing.etex

```diff
@@ -0,0 +1,266 @@
+    in
+    while true do
+        let list_ref = ref [] in (* for later fake work *)
+        for _ = 1 to 100 do
+            (* here we do some fake work *)
+            list_ref := [];
+            for _ = 1 to 10 do
+                list_ref := (Sys.opaque_identity(ref 42)) :: !list_ref
+            done;
+            Gc.full_major ();
+        done;
+        ignore(Runtime_events.read_poll cursor callbacks None);
+        Unix.sleep 1
+    done
+\end{verbatim}
+
+The next step is to compile and link the program with the runtime_events
+library. This can be done as follows:
+
+\begin{verbatim}
+       ocamlopt -I +runtime_events -I +unix unix.cmxa runtime_events.cmxa
+            example.ml -o example
+\end{verbatim}
+
+When using the {\em dune} build system, this example can be built as follows:
+
+\begin{verbatim}
+(executable
+ (name example)
+ (modules example)
+ (libraries unix runtime_events))
+\end{verbatim}
+
+Running the compiled example binary will give an output similar to:
+
+\begin{verbatim}
+Begin   explicit_gc_full_major  24086187297852
+Begin   stw_leader      24086187298594
+Begin   minor   24086187299404
+Begin   minor_global_roots      24086187299807
+End     minor_global_roots      24086187331461
+Begin   minor_remembered_set    24086187331631
+Begin   minor_finalizers_oldify 24086187544312
+End     minor_finalizers_oldify 24086187544704
+Begin   minor_remembered_set_promote    24086187544879
+End     minor_remembered_set_promote    24086187606414
+End     minor_remembered_set    24086187606584
+Begin   minor_finalizers_admin  24086187606854
+End     minor_finalizers_admin  24086187607152
+Begin   minor_local_roots       24086187607329
+Begin   minor_local_roots_promote       24086187609699
+End     minor_local_roots_promote       24086187610539
+End     minor_local_roots       24086187610709
+End     minor   24086187611746
+Begin   minor_clear     24086187612238
+End     minor_clear     24086187612580
+End     stw_leader      24086187613209
+...
+\end{verbatim}
+
+This is an example of self-monitoring, where a program explicitly starts
+runtime events and monitors itself.
+
+For external monitoring a program need not be aware of the existence of runtime
+events. Runtime events can be controlled via an environment variable
+"OCAML_RUNTIME_EVENTS_START" which, when set, will ensure that the runtime
+tracing system is started at program initialization.
+
+We could remove the "Runtime_events.start ();" from the previous example and
+instead as below to produce the same result:
+
+\begin{verbatim}
+OCAML_RUNTIME_EVENTS_START=1 ./example
+\end{verbatim}
+
+\subsubsection{s-runtime-tracing-environment-variables}{Environment variables}
+
+As we saw in the previous example, environment variables can be used to control
+aspects of the runtime events system. The following environment variables are
+available:
+
+\begin{itemize}
+  \item OCAML_RUNTIME_EVENTS_START if set will cause the runtime events system
```

Comment:
It may be useful to mention these options in the OCaml and the C consumer APIs:

* https://github.com/ocaml/ocaml/blob/trunk/otherlibs/runtime_events/runtime_events.mli
* https://github.com/ocaml/ocaml/blob/trunk/otherlibs/runtime_events/runtime_events_consumer.h

Reply:
I've documented these in the .mli and referenced them from the C header: https://github.com/ocaml/ocaml/pull/11349/commits/f01a2cdb710be358951a3a607ad6cf23a5a65054